akka-docs/src/main/paradox/interaction-patterns-typed.md

```diff
@@ -0,0 +1,103 @@
+ * If the message got lost, we will not notice
+
+## Same protocol Request-Response
+
+In many interactions a request is followed by a response back from the actor. In Akka Typed the recipient of responses has to be encoded as a field in the message itself, which the recipient can then use to send a response back. When the response message is already a part of the sending actor protocol we can simply use @scala[`ActorContext.self`]@java[`ActorContext.getSelf()`] when constructing the message.
+
+TODO sample
+
+**Scenarios where request response with tell is useful:**
+
+ * Subscribing to an actor that will send many response messages (of the same protocol) back
+ * When communicating between a parent and its children, where the protocol can be made include the messages for the interaction 
+ * ???
+
+**Problems request-response:**
+
+ * Often the response that the other actor wants to send back is not a part of the protocol the sending actor (see adapted request response or ask)
+ * It is hard to detect and that a message request was not delivered or processed (see ask)
+ * Unless the protocol already includes a way to provide context, for example a request id that is also sent in the response, it is not possible to tie an interaction to some specific context without introducing a new, separate, actor
+
+## Adapted Request-Response
+
+Very often the receiving does not, and should not be made, know of the protocol of the sending actor, and will respond with one or more messages that the sending actor cannot receive.
+
+TODO sample
+
+**Scenarios where Adapted Request-Response is useful:**
+
+ * Subscribing to an actor that will send many response messages back 
+ 
+**Problems with adapted request-response:**
+
+ * It is hard to detect and that a message request was not delivered or processed (see ask)
+ * Only one adaption can be made per response message type, if a new one is registered the old one is lost
+ * Unless the protocol already includes a way to provide context, for example a request id that is also sent in the response, it is not possible to tie an interaction to some specific context without introducing a new, separate, actor 
+
+ 
+## 1:1 Request-Response with ask between two actors
+ 
+In an interaction where there is a 1:1 mapping between a request and a response we can use @scala[`ActorContext.ask` or the implicitly provided `ActorRef.ask`]@java[`ActorContext.ask`] to interact with another actor.
+
+The interaction has two steps, first we need to construct the outgoing message, to do that we need an @scala[`ActorRef[Response]`]@java[`ActorRef<Response>`] to put as recipient in the outgoing message. The second step is to transform the `Response` or the failure to produce a response, into a message that is part of the protocol of the sending actor.
+
+TODO sample
+
```

Comment:
For scala we should explain the curried syntax in scala (replyTo: ActorRef) as second parameter list in the request message. I think we should first show ordinary `replyTo => Req(replyTo, ...)` and possibly show how the curried syntax can make it look "nicer".